### PR TITLE
Add Thanos alert examples

### PR DESCRIPTION
**Pull request by povilasv**
## Changes

Adds example alerts for Thanos for running in Kubernetes environment.
This is not meant to be suited for all environments, this is just example that you can start with and adapt to your needs.

## Verification

Here is how it looks rendered -> https://github.com/povilasv/thanos/blob/alert-examples/examples/alerts/alerts.md

### CODE REVIEW

File: docs/getting_started.md [hunk 1]
```diff
@@ -232,3 +232,7 @@ You can find one-box example with minikube [here](../kube/README.md).
 # Dashboards
 
 You can find Grafana dashboards [here](../examples/grafana/monitoring.md)
+
+# Alerts
+
+You can find Alert configuration examples [here](../examples/alerts/alerts.md)
```
Comment (Bplotka) [inline docs/getting_started.md]: `s/find Alert/find example Alert` maybe?

File: docs/getting_started.md [hunk 2]
```diff
@@ -232,3 +232,7 @@ You can find one-box example with minikube [here](../kube/README.md).
 # Dashboards
 
 You can find Grafana dashboards [here](../examples/grafana/monitoring.md)
```
Comment (Bplotka) [inline docs/getting_started.md]: `s/find Grafana/find example Grafana` maybe

File: examples/alerts/alerts.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    dashboard: COMPACTION_URL
+- alert: ThanosCompactBucketOperationsFailed
+  expr: rate(thanos_objstore_bucket_operation_failures_total{app="thanos-compact"}[5m]) > 0
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Compact bucket operations are failing
+    impact: Long term storage queries will be slower 
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: COMPACTION_URL
```
Comment (Bplotka) [inline examples/alerts/alerts.md]: what about no compaction being made in very long time?

File: examples/alerts/alerts.md [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+## Ruler
+
+For Thanos ruler we run alerts some alerts in local Prometheus:
```
Comment (Bplotka) [inline examples/alerts/alerts.md]: I don't get this sentence, I think some extra word is here

File: examples/alerts/alerts.md (hunk 3 above)
Comment (Bplotka) [reply examples/alerts/alerts.md]: plus up{compactor} == 0 + absent

Comment (Bplotka) [reply examples/alerts/alerts.md]: plus "more than one compactor is running for the same bucket (critical)" - wonder if we have metrics for that

File: examples/alerts/alerts.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+- alert: ThanosRuleIsDroppingAlerts
+  expr: rate(thanos_alert_queue_alerts_dropped_total{app="thanos-rule"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Rule is dropping alerts
+    impact: Alerts are not working
+    action: 'check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace'
+    dashboard: RULE_DASHBOARD
```
Comment (Bplotka) [inline examples/alerts/alerts.md]: We might want to check eval rule failures as well, but it is minor, we can add this in next PRs

File: examples/alerts/alerts.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  expr: thanos_sidecar_prometheus_up{name="prometheus"} == 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Sidecar cannot connect to Prometheus
+    impact: Prometheus configuration is not being refreshed
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: SIDECAR_URL
+- alert: ThanosSidecarBucketOperationsFailed
```
Comment (Bplotka) [inline examples/alerts/alerts.md]: maybe similar for store?

File: examples/alerts/alerts.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+- alert: ThanosQueryGrpcErrorRate
+  expr: rate(grpc_server_handled_total{grpc_code=~"Unknown|ResourceExhausted|Internal|Unavailable",name="prometheus"}[5m]) > 0
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Query is returning Internal/Unavailable errors 
+    impact: Grafana is not showing metrics
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: QUERY_URL
```
Comment (Bplotka) [inline examples/alerts/alerts.md]: what about query not having access to static stores? or query seeing unhealthy stores.

File: examples/alerts/alerts.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
+# Alerts
+
+Here are some example alerts configured for Kubernetes environment.
+
+## Compaction
+
+```
+- alert: ThanosCompactHalted
+  expr: thanos_compactor_halted{app="thanos-compact"} == 1
```
[the remark this answers is not in the document]
Comment (povilasv) [reply examples/alerts/alerts.md]: It all depends how you label things in Kubernetes. This is a label to pinpoint metrics from `thanos-compact`, Job wouldn't work if you are using `kubernetes_sd_discovery`. (https://prometheus.io/docs/prometheus/latest/configuration/configuration/#%3Ckubernetes_sd_config%3E)

It's trival to replace `app` to `job` or to whatever label you are using, so really don't see the point in doing that.

File: examples/alerts/alerts.md (hunk 5 above)
Comment (povilasv) [reply examples/alerts/alerts.md]: I'm not sure which metric shows that, could you point me to it?

File: examples/alerts/alerts.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (Bplotka) [reply examples/alerts/alerts.md]: so `app` is I think the default way. Job would be here `kubernetes-pods` or whatever you name your Kuberntes SD job, unless  you explicitly relabel `job` with what you have in `app` (: 

Actually we do the latter so we use `job` in our alerts. Keep in mind that these are only examples, you can use `job` in your alerts. These are only for reference.

File: examples/alerts/alerts.md (hunk 7 above)
Comment (povilasv) [reply examples/alerts/alerts.md]: I am not sure which metrics to use for that, but I typically like to alert when things really break, so I think GRPC error rate is a good proxy for that. And from there you can debug via dashboard.

File: examples/alerts/alerts.md (hunk 6 above)
Comment (povilasv) [reply examples/alerts/alerts.md]: Added in https://github.com/improbable-eng/thanos/pull/392/files#diff-fc17f1cae7faf239396714ad90fd3d06R118 :)

**Comment by povilasv** [comment]
@Bplotka Could you take a look? I've added a bunch of alerts after your review :)

File: examples/alerts/alerts.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (kfdm) [reply examples/alerts/alerts.md]: Ok, makes since :) 
Sorry again for the random drive-by question ^^;

[the remark this answers is not in the document]
Comment (povilasv) [reply examples/alerts/alerts.md]: no worries :)

File: examples/alerts/alerts.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+    impact: Long term storage queries will be slower 
+    action: Check {{ $labels.kubernetes_pod_name }} pod logs in {{ $labels.kubernetes_namespace}} namespace
+    dashboard: COMPACTION_URL
+- alert: ThanosComactionIsNotRunning
+  expr: up{app="thanos-compact"} == 0 or absent({app="thanos-compact"})
+  for: 5m
+  labels:
+    team: TEAM
+  annotations:
+    summary: Thanos Compaction is not running
```
Comment (Bplotka) [inline examples/alerts/alerts.md]: `Thanos Compactor`

**Pull request merged**